src/main/java/io/github/ust/mico/kafkafaasconnector/MessageListener.java

```diff
@@ -180,7 +199,13 @@ public void sendCloudEvent(MicoCloudEventImpl<JsonNode> cloudEvent) {
     private void sendCloudEvent(MicoCloudEventImpl<JsonNode> cloudEvent, String topic) {
         cloudEvent = this.updateRouteHistoryWithTopic(cloudEvent, topic);
         // TODO commit logic/transactions
-        kafkaTemplate.send(topic, cloudEvent);
+        if (!cloudEvent.isFilterOutNecessary(topic)){
```

Comment:
Because I think the fact if a message should be filtered or not is message related and should be handled there. This even allows adding support for different message formats in the future if we extract methods like this to a common interface. Additionally, this class is already long. Maybe we should refactor it into multiple ones.

Reply:
This class should definitely be split into receiving, calling the faas function and routing/sending.
But we will never (at least not in the foreseeable future) have another message format. And the decision is definitely a routing decision. The message related things are whether the message is a test message and whether it has a topic set to be filtered out before hitting that topic.